s390x: Add documentation for buildextend-secex (#3185)
coreos/coreos-assembler: docs/cosa/buildextend-secex.md

- The resulting image will only be encrypted with a single Hostkey, to enable firstboot
- The Hostkey will not be written to the image
- The Hostkey(s) need to be provided later during firstboot through ignition
    - Firstboot service will fail when no Hostkey is provided, as the sdboot-image can not be recreated
    - The Hostkey(s) should be written to `/etc/se-hostkeys/ibm-z-hostkey-<hostkey-name>.crt`

To facilitate this, `buildextend-secex` can take 2 mutually exclusive additional arguments: `--genprotimgvm <path>` and `--hostkey <path>`.
If none is provided, `--genprotimgvm` is used with default values.

## `--genprotimgvm <path>` (default)

Default Value: `/data.secex/genprotimgvm.qcow2`

This path is the default behavior. It assumes that the Hostkey is not directly available, but is instead only supplied through a Secure Execution protected VM.

The QEMU-Image will be build normally, however it will not run `genprotim` or `zipl`, but instead save the required input for the command to a temporary location.
After the build, it will run the provided VM. The VM is used to isolate and protect the `genprotimg` command, such that the Universal Hostkey will not be exposed.

> It will call a provided bash script before and after the genprotimg step.

Reviewer: ```suggestion
A provided bash script is called before and after the `genprotimg` command, to fullfil the following steps: 
1. Copy the required kernel, initramfs, and parmfile to the VM 
2. Move the sdboot-image to the disk
3. Call `zipl`to make the image bootable.
```

This enables us to copy the required kernel, initramfs and parmfile to the VM and afterwards move the sdboot-image to the disk, as well as calling `zipl` to make the image bootable.

## `--hostkey <path>`

> This path is intended for local development, but can be used for custom builds. It takes the path to a single Hostkey and uses it to build the image.

Reviewer: ```suggestion
This path is intended for local development, but can be used for custom builds. The path is taken to a single host key and used to build the image.
```

Author: The second path here is an actual file path. I guess it get's confusing here. I will make it more explicit.

Author: Addressed in  4cdd78a

> Instead of running `genprotimg` and `zipl` in a separate VM, they will be run during the build process. Otherwise the build is identical to the `--genprotimgvm`.

Reviewer: ```suggestion
Instead of running `genprotimg` and `zipl` in a separate VM, they run during the build process. Otherwise, the build is identical to the `--genprotimgvm`.
```

> This also means that it is still assumed that the Hostkey will be provided via ignition during firstboot.

Reviewer: ```suggestion
Note: It is still assumed that the host key is provided via Ignition during firstboot.
```
